homeassistant/components/huisbaasje/config_flow.py

```diff
@@ -0,0 +1,94 @@
+"""Config flow for Huisbaasje integration."""
+import logging
+
+from huisbaasje import Huisbaasje, HuisbaasjeConnectionException, HuisbaasjeException
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_ID, CONF_PASSWORD, CONF_USERNAME
+
+from .const import DOMAIN  # pylint: disable=unused-import
+
+_LOGGER = logging.getLogger(__name__)
+
+DATA_SCHEMA = vol.Schema(
+    {vol.Required(CONF_USERNAME): str, vol.Required(CONF_PASSWORD): str}
+)
+
+
+class HuisbaasjeConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Huisbaasje."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_CLOUD_POLL
+
+    async def async_step_user(self, user_input=None):
+        """Handle a flow initiated by the user."""
+        if user_input is None:
+            return await self._show_setup_form(user_input)
+
+        errors = {}
+
+        try:
+            user_id = await self._validate_input(user_input)
+
+            _LOGGER.info("Input for Huisbaasje is valid!")
+
+            # Set user id as unique id
+            existing_config_entry = await self.async_set_unique_id(user_id)
+
+            # Abort if entry already exists
+            if existing_config_entry is not None:
+                return self.async_abort(reason="already_configured")
+            if await self._entry_exists(user_id):
+                return self.async_abort(reason="already_configured")
```

Comment:
This is already built into the class that is inherited from here.

```suggestion
            # Set user id as unique id
            await self.async_set_unique_id(user_id)
            self._abort_if_unique_id_configured()
```

See: <https://developers.home-assistant.io/docs/config_entries_config_flow_handler#unique-ids>

Reply:
Sure. But isnt _abort_if_unique a private method? Somebody pointed it out to me so I changed it to what it is here

Reply:
It is, but it's private to its current context. (If you are more in Java terms, it would be called protected).